compute_sdk/globus_compute_sdk/sdk/login_manager/tokenstore.py

```diff
@@ -36,22 +36,31 @@ def invalidate_old_config() -> None:
 
 
 def ensure_compute_dir() -> pathlib.Path:
+    user_dirname_str = os.getenv("GLOBUS_COMPUTE_USER_DIR")
+    if user_dirname_str:
+        user_dirname = pathlib.Path(user_dirname_str)
+        if user_dirname.is_dir():
+            return user_dirname
+        elif user_dirname.is_file():
+            raise FileExistsError(f"{user_dirname} must be a directory, not a file")
+        else:
+            raise FileNotFoundError(
+                f"{user_dirname} could not be found on the file system"
+            )
```

Comment:
Below, we make the directory if it doesn't exist.  Why not fold this into the main logic that already exists?  Something like:
```python
legacy_dirname = _home() / ".funcx"
dirname = os.getenv("GLOBUS_COMPUTE_USER_DIR", _home() / ".globus_compute")
dirname = pathlib.Path(dirname)
```

Reply:
Talking to Reid and mainly the renaming/symlink logic is functionally different, but I have to think on what the 'expected' behavior is from the user POV.

Reply:
I separated it out for a few reasons:

1. Avoid the workflow for renaming the legacy dir and creating a symlink.
2. Avoid attempting to create a dir with insufficient permissions (e.g., `/root/mydir`). I'm open to changing this.
3. Slightly different error messages.

Reply:
Spoke with Kevin offline and decided to move forward with the following:

1. Modify the existing logic to skip the symlink workflow if a custom user dir is defined.
2. Attempt to create the specified user dir if it doesn't exist. If it's specified parent dir is restricted, an exception will be raised then re-raised as a `ClickException` in the endpoint CLI.
3. The changes above make it unnecessary to use different error messages.